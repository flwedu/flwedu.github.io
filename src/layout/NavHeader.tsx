export function NavHeader() {
  return (
    <nav>
      <ul>
        <li>
          <a href="#about">Home</a>
        </li>
        <li>
          <a href="#technologies">Tecnologias</a>
        </li>
        <li>
          <a href="#projects">Projetos</a>
        </li>
        <li>
          <a href="#">Links &dtrif;</a>
          <ul className="dropdown">
            <li>
              <a href="https://github.com/flwedu" target="_blank">
                GitHub{" "}
              </a>
            </li>
            <li>
              <a href="https://twitter.com/emprestavel" target="_blank">
                Twitter
              </a>
            </li>
            <li>
              <a href="https://www.linkedin.com/in/edu-aquino/" target="_blank">
                LinkedIn
              </a>
            </li>
            <li>
              <a href="https://www.codewars.com/users/flwedu" target="_blank">
                Codewars
              </a>
            </li>
            <li>
              <a href="https://www.hackerrank.com/edutraquino" target="_blank">
                HackerRank
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </nav>
  );
}
