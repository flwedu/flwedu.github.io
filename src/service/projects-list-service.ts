import { Project } from "../components/ProjectCard";

export class ProjectsListService {
  constructor(private filePath: string) {}

  async getList(): Promise<Project[]> {
    const list = await (await fetch(this.filePath)).json();
    const projects: Project[] = list.map((el: any) => el as Project);
    return projects;
  }
}
