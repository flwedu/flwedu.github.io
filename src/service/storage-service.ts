export type Settings = {
  darkmode: boolean;
  language: "en" | "pt-br";
};

export class StorageService {
  constructor() {}

  save(settings: Settings) {
    const json = JSON.stringify(settings);
    localStorage.setItem("settings", json);
  }

  load() {
    const json = localStorage.getItem("settings");
    if (json) {
      const settings = JSON.parse(json) as Settings;
      return settings;
    }
    const settings: Settings = { darkmode: false, language: "pt-br" };
    return settings;
  }
}
