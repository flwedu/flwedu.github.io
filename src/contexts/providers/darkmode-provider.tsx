//@ts-nocheck
import { ReactNode, createContext, useEffect, useState } from "react";
import { StorageService } from "../../service/storage-service";

type DarkModeContextType = {
  darkmode: boolean;
  toggleFn: () => void;
};

const DarkModeContext = createContext<DarkModeContextType | null>(null);

const DarkModeContextProvider = (props: {
  children: ReactNode;
  storageService: StorageService;
}) => {
  const initialDarkMode = props.storageService.loadItem("darkmode") as boolean;

  const [darkmode, setDarkMode] = useState<boolean>(initialDarkMode);

  // Check if dark mode is enabled at initialization
  if (darkmode) document.documentElement.classList.add("dark");

  useEffect(() => {
    props.storageService.saveItem("darkmode", darkmode);
  }, [darkmode]);

  // Function that is used to toggle dark mode
  const toggleFn = () => {
    document.documentElement.classList.toggle("dark");
    setDarkMode(document.documentElement.classList.contains("dark"));
  };

  return (
    <DarkModeContext.Provider value={{ darkmode, toggleFn }}>
      {props.children}
    </DarkModeContext.Provider>
  );
};

export { DarkModeContext, DarkModeContextProvider };
