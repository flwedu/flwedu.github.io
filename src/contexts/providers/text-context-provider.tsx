//@ts-nocheck
import { createContext, useEffect, useState } from "react";
import textsSource from "../../../static/res/texts.json";
import { StorageService } from "../../service/storage-service";

type TextSource = {
  language: "en" | "pt-br";
  emoji: "ðŸ‡ºðŸ‡¸" | "ðŸ‡§ðŸ‡·",
    buttons: {
      toDarkMode: string,
      toLightMode: string,
      language: string
    },
  navbar: {
    home: string;
    technologies: string;
    projects: string;
    links: string;
  };
  sections: {
    about: {
      h1: string;
      p: string;
    };
    technologies: {
      h1: string;
      p: string;
    };
    projects: {
      h1: string;
      p: string;
    };
  };
};

type TextContextType = {
  text: TextSource;
  language: "en" | "pt-br";
  toggleFn: () => void;
};

const TEXT_EN = textsSource["en"];
const TEXT_PTBR = textsSource["pt-br"];

const TextContext = createContext<TextContextType | null>(null);

const TextContextProvider = (props: {
  children: JSX.Element;
  storageService: StorageService;
}) => {
  const initialLanguage = props.storageService.loadItem("language");
  const [language, setLanguage] = useState(initialLanguage);
  const [text, setText] = useState(textsSource[initialLanguage]);

  // saving new preferences when text is changed
  useEffect(() => {
    props.storageService.saveItem("language", text.language);
    setLanguage(text.language);
  }, [text]);

  // Toggle language selection, useEffect function above this is triggered
  const toggleFn = () => {
    if (text.language == "pt-br") {
      return setText(TEXT_EN);
    }
    return setText(TEXT_PTBR);
  };

  return (
    <TextContext.Provider value={{ text, toggleFn, language }}>
      {props.children}
    </TextContext.Provider>
  );
};

export { TextContext, TextContextProvider };
