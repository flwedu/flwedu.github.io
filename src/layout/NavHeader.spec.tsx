import { render } from "@testing-library/react";
import { HashRouter } from "react-router-dom";
import { TextContextProvider } from "../contexts/providers/text-context-provider";
import { StorageService } from "../service/storage-service";
import { NavHeader } from "./NavHeader";

describe("NavHeader component test", () => {
  test("Rendered component should match the snapshot", () => {
    const storageService = new StorageService();
    const { container } = render(
      <TextContextProvider storageService={storageService}>
        <HashRouter>
          <NavHeader />
        </HashRouter>
      </TextContextProvider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});

export {};
