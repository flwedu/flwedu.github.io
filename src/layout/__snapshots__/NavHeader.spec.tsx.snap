// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavHeader component test Rendered component should match the snapshot 1`] = `
<nav
  class="sticky container flex flex-row mx-auto justify-center content-center top-0 max-w-3xl h-12 bg-white border-gray-200 dark:bg-gray-800 rounded shadow-md"
>
  <ul
    class="flex flex-row justify-center items-center gap-5"
  >
    <li>
      <a
        href="#/"
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="#/technologies"
      >
        Tecnologias
      </a>
    </li>
    <li>
      <a
        href="#/projects"
      >
        Projetos
      </a>
    </li>
    <li>
      <a
        href="#/links"
      >
        Links
      </a>
    </li>
  </ul>
</nav>
`;
